### ISSUE TITLE
Improve object latency metric `thanos_objstore_bucket_operation_duration_seconds`

### ISSUE THREAD

**Issue by bwplotka** [thanos-io/thanos#1695]
**Thanos, Prometheus and Golang version used**: "master-2019-10-29-b7f3ac9e"

**What happened**:

Saw this during benchmarking.

Let's say we want to know if object storage caused any major latency during request or store GW startup:
Let's say we use histogram, eg: `histogram_quantile(0.95, sum(rate(thanos_objstore_bucket_operation_duration_seconds_bucket[1m])) by (operation, instance, pod, le))`

For certain operations, however, we return incorrect data e.g:
![image](https://user-images.githubusercontent.com/6950331/67796937-c898fe00-fa78-11e9-9650-01ba5d03dec0.png)

This is during Store GW initialization. Overall the whole init took 10s. With 18 test blocks in the bucket, tail latency is most likely NOT 10s against object storage. Especially for such small files, I was benchmarking.

**What you expected to happen**:

Metric should show actual latency that was caused by downloading something from object storage.
Additionally, we don't have `iter` operation included - for the same reason. IT would be nice to have.

### Problem:

Streaming. Measuring latency for certain operations while streaming in multicore system has to be carefully done. Currently, we measure bucket operation latency inside the reader: https://github.com/thanos-io/thanos/blob/64af185e73978986c7955d2b3e8ad49e33bb099c/pkg/objstore/objstore.go#L346 and this, unfortunately, accounts for non object storage operational latency (e.g caller thread blocked on something). This is what is happening on startup as we use some concurrency (e.g 20) so overall the full read operation can take between 5s-10s however, the actual fetch from storage took in sum maybe ms...  Let's fix it!

I would propose just measuring the time spent when invoking `Read`. And sum all of it together to have proper fetch latency. The opposite should be done on upload.

The same value we should log/trace.

NOTE: Exactly the same is happening for `gRPC Series` request latency reported by GRPC. We stream a lot and server might just wait on client to receive it. This should not be accounted for server latency

### AC:

* We have `thanos_objstore_bucket_operation_duration_seconds` per `Iter` operation as well.
* `thanos_objstore_bucket_operation_duration_seconds` accounts only for object storage latency (!)
